integration_tests/run_pyspark_from_build.sh

```diff
@@ -70,6 +70,21 @@ else
         # to still launch a spark application.
         TEST_PARALLEL=`nvidia-smi --query-gpu=memory.free --format=csv,noheader | awk '{if (MAX < $1){ MAX = $1}} END {print int(MAX / (2.3 * 1024)) - 1}'`
         echo "AUTO DETECTED PARALLELISM OF $TEST_PARALLEL"
+
+        # adjust TEST_PARALLEL according to cpu cores and free memory
+        cpu_cores=$(nproc --all)
+        free_mem_mib=$(awk '/MemFree/ { printf "%d\n", $2/1024 }' /proc/meminfo)
+        max_parallel_for_cpu_cores=$(python -c "print($cpu_cores - 1)")
```

Comment:
Nit: Firing up Python here for some simple math seems like overkill, can use bash's [arithmetic evaluation](https://tldp.org/LDP/abs/html/arithexp.html) to do this directly in the shell.
 ```suggestion
        max_parallel_for_cpu_cores=$(($cpu_cores - 1))
```